src/System/Validation/Code.cs

```diff
@@ -55,6 +55,29 @@ public static IEnumerable<T> ExpectsAllNotNull<T>([ValidatedNotNull] IEnumerable
 		}
 
 
+		/// <summary>
+		/// Checks the enumeable argument and throws an exception if it is null, contains no values, or contains any null values
+		/// </summary>
+		/// <remarks>Be careful to not pass enumerables that can be enumerated only once</remarks>
+		/// <typeparam name="T">Type of the enumerable</typeparam>
+		/// <param name="argumentValue">The argument value.</param>
+		/// <param name="argumentName">Name of the argument.</param>
+		/// <param name="tagId">Tag Id to log, leave null if no logging is needed.</param>
+		/// <exception cref="ArgumentException">Thrown if any argument  <paramref name="argumentValue"/> element is null.</exception>
+		/// <exception cref="ArgumentNullException">Thrown if the supplied argument <paramref name="argumentValue"/> is null.</exception>
+		public static IEnumerable<T> ExpectsAnyAndAllNotNull<T>([ValidatedNotNull] IEnumerable<T> argumentValue, string argumentName, uint? tagId) where T : class
```

Comment:
I feel that the name might be a bit confusing, I was reading it like (any and all) not null, what? It might be clear enough to call it ExpectsAllNotNull?